### PR TITLE
 IOW-20 - CSRF & other acunetix prevention 

### PR DESCRIPTION
**Pull request by dsteinich**
Before making a pull request
----------------------------

- [ ] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java [hunk 1]
```diff
@@ -57,68 +58,47 @@ public BaseController(LookupDao inLookupDao, StreamingDao inStreamingDao, Naviga
...
+			String sessionId = getSessionId(parameterMap, response);
+			if (null != sessionId) {
+				parameterMap.put(StreamingDao.SESSION_ID, sessionId);
 				addContentHeader(response);
-				streamResults(transformer, BaseDao.FLOW_LINES, parameterMap);
+				streamResults(transformer, BaseDao.FLOW_LINES_LEGACY, parameterMap);
 			}
-
-		} catch (Exception e) {
-			LOG.error(e.getLocalizedMessage());
```
Comment (dsteinich) [inline src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java]: This error handling method was exposing implementation details (things like sql errors & code) and preventing the global exception handling from working.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -21,3 +21,22 @@ nldi:
   displayProtocol: ${nldiProtocol}
   displayHost: ${nldiHost}
   displayPath: ${nldiPath:}
+
```
Comment (dsteinich) [inline src/main/resources/application.yml]: This renames the Actuator endpoints to hide them and only allows the heath and info ones.

File: src/main/java/gov/usgs/owi/nldi/springinit/WebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package gov.usgs.owi.nldi.springinit;
+
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
```
Comment (dsteinich) [inline src/main/java/gov/usgs/owi/nldi/springinit/WebSecurityConfig.java]: Unfortunately, spring security is needed for the csrf functionality, forcing us to permit all access to all endpoints.

File: src/test/java/gov/usgs/owi/nldi/controllers/CharacteristicsControllerIT.java [hunk 4]
```diff
@@ -1,94 +1,88 @@
...
-import org.springframework.web.servlet.config.annotation.EnableWebMvc;
 
 @EnableWebMvc
-@AutoConfigureMockMvc
-@SpringBootTest(webEnvironment=WebEnvironment.MOCK,
-		classes={DbTestConfig.class, SpringConfig.class, 
-		CharacteristicsController.class, LookupDao.class, StreamingDao.class,
-		Navigation.class, NavigationDao.class, Parameters.class, 
-		ConfigurationService.class, LogService.class, LogDao.class})
+@SpringBootTest(webEnvironment=WebEnvironment.RANDOM_PORT)
```
Comment (dsteinich) [inline src/test/java/gov/usgs/owi/nldi/controllers/CharacteristicsControllerIT.java]: Another side effect of security - need to change the IT tests to fire up the entire servlet context.

File: README.md [hunk 5]
```diff
@@ -6,10 +6,10 @@
 ## Public API
 The services are accessed via an http GET request. All navigation output is generated as GeoJSON ("application/vnd.geo+json")
 
-#### The root is {host}/nldi.
+#### The root is {host}/nldi/api.
 Both a test and a production are exposed to the public:
-__https://cida-test.er.usgs.gov/nldi__ is the test root.
-__https://cida.usgs.gov/nldi__ is the production root.
+__https://labs.waterdata.usgs.gov/nldi/api__ is the test root.
```
Comment (mbucknell) [inline README.md]: This looks a little strange since both the test and public urls are the same.

Comment (dsteinich) [reply README.md]: Unclear what the new urls are - not anything cida...

**Pull request merged**